parser.py

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/python3
+# parser.py
+
+
+def parse(text):
```

Comment:
Instead of reading in the entire file at once, can you change `parse` to use buffering instead? The data files can be huge. With buffering, things like removing empty lines as you see them will also be faster than operating on the entire memory at once.